src/popup/components/Modals/TransferSendBase.vue

```diff
@@ -0,0 +1,113 @@
+<template>
+  <Modal
+    class="transfer-send-base"
+    has-close-button
+    from-bottom
+    :body-without-padding-bottom="currentStep === TRANSFER_SEND_STEPS.form"
+    @close="$emit('close')"
+  >
+    <div class="relative">
+      <transition name="fade-between">
+        <slot name="content" />
+      </transition>
+    </div>
+    <template #footer>
+      <BtnMain
+        v-if="showEditButton"
+        variant="muted"
+        :text="$t('common.edit')"
+        class="button-action-secondary"
+        data-cy="edit"
+        @click="editTransfer"
+      />
+      <BtnMain
+        class="button-action-primary"
+        :disabled="!isConnected || primaryButtonDisabled"
+        :icon="(showSendButton && hideArrowSendIcon) ? ArrowSendIcon : null"
+        :text="primaryButtonText"
+        data-cy="next-step-button"
+        @click="proceedToNextStep"
+      />
+    </template>
+  </Modal>
+</template>
+
+<script lang="ts">
+import {
+  computed,
+  defineComponent,
+  PropType,
+} from 'vue';
+import type {
+  TransferSendStep,
+} from '@/types';
+import {
+  TranslateResult,
+  useI18n,
+} from 'vue-i18n';
+
+import { useGetter } from '@/composables/vuex';
+import { TRANSFER_SEND_STEPS } from '@/constants';
+import Modal from '../Modal.vue';
+import BtnMain from '../buttons/BtnMain.vue';
+import ArrowSendIcon from '../../../icons/arrow-send.svg?vue-component';
+
+export default defineComponent({
+  name: 'TransferSendBase',
+  components: {
+    Modal,
+    BtnMain,
+  },
+  props: {
+    editTransfer: { type: Function, default: () => null },
+    proceedToNextStep: { type: Function, default: () => null },
+    customPrimaryButtonText: { type: String as PropType<TranslateResult | string>, default: '' },
```

Comment:
No need to use `TranslateResult`. With the newer version of i18n the translate result is treated as a string.